src/lib/links/customSelector.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+
+interface Option {
+  value: string;
+}
+
+interface CustomSelectProps {
+  options: Option[];
+  selectedValue: string;
+  onChange: (value: string) => void;
+}
+
+const CustomSelect: React.FC<CustomSelectProps> = ({
```

Comment:
Probably you could make use of `floating-ui` library which is already installed. It provides all utils necessary for good ux via mouse, keyboard, and screen readers.
You can view the dropdown example: https://floating-ui.com/docs/react-examples